targets/XPlatCppSdk/source/build/Android/UnitTest/TestAppAndroid.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <jni.h>
+#include <android/log.h>
+#include <string>
+#include <fstream>
+
+#include "TestAppPch.h"
+#include "TestApp.h"
+
+//#define USE_EXTERNAL_JSON_FILE
+
+#ifdef USE_EXTERNAL_JSON_FILE
+#warning "Replace below file name(except ext) with yours, and DO NOT SHARE IT."
+static const char* c_jsonFileName = "YOUR_JSON_FILE_NAME";
+#else // USE_EXTERNAL_JSON_FILE
+#warning "Replace below information with yours, and DO NOT SHARE IT."
+static const char* c_titleId = "YOUR_TITLE_ID";
+static const char* c_developerSecretKey = "YOUR_DEVELOPER_SECRET_KEY";
+static const char* c_userEmail = "YOUR_EMAIL";
+#endif // USE_EXTERNAL_JSON_FILE
+
+namespace PlayFabUnit
+{
+    bool TestApp::LoadTitleDataJson(std::shared_ptr<char*>& testDataJson, size_t& testDataJsonLen)
+    {
+#ifdef USE_EXTERNAL_JSON_FILE
+        NSString *filePath = [[NSBundle mainBundle] pathForResource:c_jsonFileName ofType:@"json"];
```

Comment:
This is my mistake I refered iOS implementation. This code should be removed or implemented for Android.

---
In reply to: [275535741](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r275535741) [](ancestors = 275535741)

Reply:
Actually I've removed this part for Android, leaving it empty and specifying TODO comment for future, if required. Looks like I somehow have not submitted it, so it's kind of my mistake :)
Raymond, I'd recommend you do the same, just leave it blank with TODO because we don't have proper scenario for packaging Android test app and test it right now.

Reply:
Added a backlog [https://microsoft.visualstudio.com/Xbox/_workitems/edit/21256721](https://microsoft.visualstudio.com/Xbox/_workitems/edit/21256721)

---
In reply to: [275923271](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r275923271) [](ancestors = 275923271)